specs/data-grid/data-grid.spec.md

```diff
@@ -0,0 +1,301 @@
+# Data Grid
+
+## Overview
+
+The `data grid` component enables authors to display an array of data in a tabular layout.  The layout can include a "header" region where a title can be displayed for each column.  
+
+![](./images/grid.png)
+
+`Data grid` actually consists of a number of components that work together:
+- `<fast-data-grid>`:  The top level container element
+- `<fast-data-grid-header>`: Displays column titles
+- `<fast-data-grid-header-cell>`: Displays the title for a single column of cells
+- `<fast-data-grid-row>`: Displays a single row of data associated with a single record
+- `<fast-data-grid-cell>`: Displays a single cell of data within a row
+
+### Use Cases
+Any time an author wants to display tabular data.
+
+### Non-goals
+- The initial version of the component will not support virtualization or pagination to display large data sets but should be architected to support both in the future.
+  
+### Features
+- Generate a data grid layout based on provided data.
+- Multiple customization points to control the grid display
+- Manages keyboard navigation across the grid
+
+### Risks and Challenges
+- It is not yet clear how cells with focusable children work.
+- Unique identifiers are not required per row/data item, this may cause an unstable relationships between data rows and component representations.
+
+### Prior Art/Examples
+
+[FAST React](https://github.com/microsoft/fast-react/tree/master/packages/react/fast-components-react-msft/src/data-grid)
+
+[Infragistics](https://www.infragistics.com/products/ignite-ui-web-components/web-components/components/data-grid.html#:~:text=%20Data%20Grid%20Overview%20%201%20Demo.%20,Components%20data%20grid%20to%20the%20above...%20More%20)
+
+[Smart](https://www.htmlelements.com/docs/grid/)
+
+[Vaadin](https://vaadin.com/components/vaadin-grid)
+
+[Open UI](https://open-ui.org/components/table)
+
+
+---
+
+## Design
+
+The Fast `Data grid` components enable a high degree of customizability. In addition to being able to set the base styles of the components authors can choose the templates applied to grid and header cells on a per column basis using the properties of `DataGridColumn`. Additionally, authors can specify the templates used during the creation of rows and cells from data sources through the item template properties of the grid and row components (`rowItemTemplate` and `cellItemTemplate`).  Finally, authors can add elements they create and manage themselves through the various element slots.
+
+### API
+
+#### The DataGridColumn interface
+
+Most `data grid` components use the `DataGridColumn` interface.  A `DataGridColumn` is an object that describes the properties of a single column in the grid as follows:
```

Comment:
Since backticks are used for code I would instead place emphasis on "data grid" using the double stars to make it bold.

```suggestion
Most **data grid** components use the `DataGridColumn` interface.  A `DataGridColumn` is an object that describes the properties of a single column in the grid as follows:
```